### PR TITLE
win: move child_stdio_buffer out of uv_process_t

### PR DESCRIPTION
**Pull request by santigimeno**

### CODE REVIEW

File: include/uv/win.h [hunk 1]
```diff
@@ -606,7 +606,7 @@ typedef struct {
   struct uv_process_exit_s {                                                  \
     UV_REQ_FIELDS                                                             \
   } exit_req;                                                                 \
-  BYTE* child_stdio_buffer;                                                   \
+  void* unusued; /* TODO: retained for ABI compat; remove this in v2.x. */    \
```
Comment (bnoordhuis) [inline include/uv/win.h:609]:
```suggestion
  void* unused; /* TODO: retained for ABI compat; remove this in v2.x. */    \
```
Or was that an intentional typo? :-)

File: src/win/process.c [hunk 2]
```diff
@@ -1040,7 +1040,8 @@ int uv_spawn(uv_loop_t* loop,
     }
   }
 
-  err = uv__stdio_create(loop, options, &process->child_stdio_buffer);
+  child_stdio_buffer = NULL;
```
Comment (bnoordhuis) [inline src/win/process.c:1043]: This needs to be moved to before the first `goto done`, before line 972 if I count correctly.

File: include/uv/win.h (hunk 1 above)
Comment (santigimeno) [reply include/uv/win.h]: Yeah, I love inventing new words.

File: src/win/process.c (hunk 2 above)
Comment (santigimeno) [reply src/win/process.c]: Thanks for catching this up, I was a bit careless here.

**Comment by santigimeno** [comment]
Updated the PR with Ben's comments. Thanks!

**Review by bnoordhuis — APPROVED** [review]

**Review by vtjnash — APPROVED** [review]

**Pull request merged**